op-node/cmd/interop/interop.go

```diff
@@ -0,0 +1,312 @@
+			L2ChainIDs:       cliCtx.Uint64Slice(l2ChainIDsFlag.Name),
+			GenesisTimestamp: cliCtx.Uint64(timestampFlag.Name),
+		}
+		if recipe.GenesisTimestamp == 0 {
+			recipe.GenesisTimestamp = uint64(time.Now().Unix() + 5)
+		}
+		mnemonic := strings.TrimSpace(cliCtx.String(mnemonicFlag.Name))
+		if mnemonic == devkeys.TestMnemonic {
+			logger.Warn("Using default test mnemonic!")
+		}
+		keys, err := devkeys.NewMnemonicDevKeys(mnemonic)
+		if err != nil {
+			return fmt.Errorf("failed to setup dev keys from mnemonic: %w", err)
+		}
+		worldCfg, err := recipe.Build(keys)
+		if err != nil {
+			return fmt.Errorf("failed to build deploy configs from interop recipe: %w", err)
+		}
+		if err := worldCfg.Check(logger); err != nil {
+			return fmt.Errorf("invalid deploy configs: %w", err)
+		}
+		artifactsDir := cliCtx.String(artifactsDirFlag.Name)
+		af := foundry.OpenArtifactsDir(artifactsDir)
+		var srcFs *foundry.SourceMapFS
+		if cliCtx.IsSet(foundryDirFlag.Name) {
+			srcDir := cliCtx.String(foundryDirFlag.Name)
+			srcFs = foundry.NewSourceMapFS(os.DirFS(srcDir))
+		}
+		worldDeployment, worldOutput, err := interopgen.Deploy(logger, af, srcFs, worldCfg)
+		if err != nil {
+			return fmt.Errorf("failed to deploy interop dev setup: %w", err)
+		}
+		outDir := cliCtx.String(outDirFlag.Name)
+		// Write deployments
+		{
+			deploymentsDir := filepath.Join(outDir, "deployments")
+			l1Dir := filepath.Join(deploymentsDir, "l1")
+			if err := writeJson(filepath.Join(l1Dir, "common.json"), worldDeployment.L1); err != nil {
+				return fmt.Errorf("failed to write L1 deployment data: %w", err)
+			}
+			if err := writeJson(filepath.Join(l1Dir, "superchain.json"), worldDeployment.Superchain); err != nil {
+				return fmt.Errorf("failed to write Superchain deployment data: %w", err)
+			}
+			l2sDir := filepath.Join(deploymentsDir, "l2")
+			for id, dep := range worldDeployment.L2s {
+				l2Dir := filepath.Join(l2sDir, id)
+				if err := writeJson(filepath.Join(l2Dir, "addresses.json"), dep); err != nil {
+					return fmt.Errorf("failed to write L2 %s deployment data: %w", id, err)
+				}
+			}
+		}
+		// write genesis
+		{
+			genesisDir := filepath.Join(outDir, "genesis")
+			l1Dir := filepath.Join(genesisDir, "l1")
+			if err := writeJson(filepath.Join(l1Dir, "genesis.json"), worldOutput.L1.Genesis); err != nil {
+				return fmt.Errorf("failed to write L1 genesis data: %w", err)
+			}
+			l2sDir := filepath.Join(genesisDir, "l2")
+			for id, dep := range worldOutput.L2s {
+				l2Dir := filepath.Join(l2sDir, id)
+				if err := writeJson(filepath.Join(l2Dir, "genesis.json"), dep.Genesis); err != nil {
+					return fmt.Errorf("failed to write L2 %s genesis config: %w", id, err)
+				}
+				if err := writeJson(filepath.Join(l2Dir, "rollup.json"), dep.RollupCfg); err != nil {
+					return fmt.Errorf("failed to write L2 %s rollup config: %w", id, err)
+				}
+			}
+		}
+		return nil
+	},
+}
+
+func writeJson(path string, content any) error {
```

Comment:
I'm surprised we don't have some utility for this already in the monorepo? Could we chuck this into a utility package?

Reply:
there is, but it was incomplete, it doesn't allow you to just write a basic file. added some missing utils, and now sharing the same op-service json-writing code.